public/course_lecture.php

```diff
@@ -0,0 +1,211 @@
+
+    foreach ($course_sections as $section) {
+        try {
+            $course_section_student = CourseSectionStudent::includes("user")->where(array('section_id' => $section->id));
+            foreach ($course_section_student as $student){
+                array_push($course_students, $student->user);
+            }
+            
+        } catch (PDOException $error) {
+            echo "<br>" . $error->getMessage();
+        }
+    }
+
+    if ($course_students && count($course_students)) { ?>
+    <table>
+            <thead>
+                <tr>
+                    <th>#</th>
+                    <th>First Name</th>
+                    <th>Last Name</th>
+                    <th>Email Address</th>
+                    <th>Created At</th>
+                </tr>
+            </thead>
+            <tbody>
+        <?php foreach ($course_students as $row) { ?>
+            <tr>
+                <td><?php echo escape($row->id); ?></td>
+                <td><?php echo escape($row->first_name); ?></td>
+                <td><?php echo escape($row->last_name); ?></td>
+                <td><?php echo escape($row->email); ?></td>
+                <td><?php echo escape($row->created_at);  ?> </td>
+            </tr>
+        <?php } ?>
+        </tbody>
+    </table>
+    <?php } else { ?>
+        <blockquote>No Students found for this course.</blockquote>
+    <?php }?>
+
+    <h2>Teams</h2>
+    <?php if ($teams && count($teams)) { ?>
+    
+        <?php foreach ($teams as $row) { ?>
+           <h3><?php echo "Team # ".$row->id ?></h3>
+           <table>
+                <thead>
+                    <tr>
+                        <th>#</th>
+                        <th>First Name</th>
+                        <th>Last Name</th>
+                        <th>Email Address</th>
+                        <th>Created At</th>
+                    </tr>
+                </thead>
+                <tbody>
+                <?php $students_in_team = TeamMember::includes("user")->where(array('team_id' => $row->id)); 
+                foreach ($students_in_team as $student) {?>
+                <tr>
+                    <td><?php echo escape($student->user_id); ?></td>
+                    <td><?php echo escape($student->user->first_name); ?></td>
+                    <td><?php echo escape($student->user->last_name); ?></td>
+                    <td><?php echo escape($student->user->email); ?></td>
+                    <td><?php echo escape($student->user->created_at);  ?> </td>
+                </tr>
+                <?php }?>
+                </tbody>
+                </table>
+
+        <?php } ?>
+        </tbody>
+    </table>
+    <?php } else { ?>
+        <blockquote>No teams found for this course.</blockquote>
+    <?php }?>
+
+    <a href="team_creation.php?id=<?php echo $course_page_id ?>">Create new teams</a> 
+
+<?php } ?>
+
+<h2>Marked entities</h2>
+<?php
+if ($marked_entities && count($marked_entities)) { ?>
```

Comment:
im curious, whats the purpose of using count() here? if youre checking if the array is empty, wouldnt empty() fit the bill? will count() return a count of 0 if the array is empty? if so, i think the codeblock would execute right?

Reply:
If I'm honest, I've done this because I saw it done elsewhere :S

Reply:
I think this is fine too, 0 is not truthy (probably)